beacon-chain/rpc/validator/proposer.go

```diff
@@ -353,6 +345,48 @@ func (vs *Server) canonicalEth1Data(ctx context.Context, beaconState *stateTrie.
 	return canonicalEth1Data, canonicalEth1DataHeight, nil
 }
 
+func (vs *Server) depositTrie(ctx context.Context, canonicalEth1DataHeight *big.Int) (*trieutil.SparseMerkleTrie, error) {
+	var depositTrie *trieutil.SparseMerkleTrie
+
+	var finalizedDeposits *depositcache.FinalizedDeposits
+	if featureconfig.Get().EnableFinalizedDepositsCache {
```

Comment:
maybe just do the whole operation here under this conditional when the feature flag is enabled. Otherwise it is hard to understand what exactly is happening. So:
```suggestion
	if featureconfig.Get().EnableFinalizedDepositsCache {
	finalizedDeposits = vs.DepositFetcher.FinalizedDeposits(ctx)
	   depositTrie = finalizedDeposits.Deposits

		upToEth1DataDeposits := vs.DepositFetcher.NonFinalizedDeposits(ctx, canonicalEth1DataHeight)
		insertIndex := finalizedDeposits.MerkleTrieIndex + 1
		for _, dep := range upToEth1DataDeposits {
			depHash, err := ssz.HashTreeRoot(dep.Data)
			if err != nil {
				return nil, errors.Wrap(err, "could not hash deposit data")
			}
			depositTrie.Insert(depHash[:], int(insertIndex))
			insertIndex++
		}
		return depositTrie, nil

```